lib/active_job/queue_adapters/sidekiq_adapter.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+# Define a new class whose name is shorter than from Active Job.
+module Sidekiq
+  module ActiveJob
+    # @api private
+    class Wrapper
+      include Sidekiq::Job
+
+      def perform(job_data)
+        ::ActiveJob::Base.execute(job_data.merge("provider_job_id" => jid))
+      end
+    end
+  end
+end
+
+module ActiveJob
+  module QueueAdapters
+    # Explicitly remove the implementation existing in older rails'.
+    remove_const(:SidekiqAdapter) if defined?(:SidekiqAdapter)
+
+    # Sidekiq adapter for Active Job
+    #
+    # To use Sidekiq set the queue_adapter config to +:sidekiq+.
+    #
+    #   Rails.application.config.active_job.queue_adapter = :sidekiq
+    class SidekiqAdapter
+      # Defines whether enqueuing should happen implicitly to after commit when called
+      # from inside a transaction.
+      # @api private
+      def enqueue_after_transaction_commit?
+        true
```

Comment:
@mperham does it mean we don't need `Sidekiq.transactional_push! (after_commit_everywhere)` in Rails 7.2 and newer? Should https://github.com/sidekiq/sidekiq/wiki/Advanced-Options#transactional-push be updated?

Reply:
I'm assuming that the Rails support only applies to Active Jobs. Sidekiq::Jobs still need after_commit_everywhere because Sidekiq has no required dependencies on Rails, SQL databases or transactions.